### PR TITLE
✨Flags for PROS 4 Beta Templates

### PR DESCRIPTION
**Pull request by omegaStag**
#### Summary:
Adds a beta depot to the remote depots (currently only contains a 3.7.3 kernel template labeled as 4.0.0), adds ``--pros-4`` flags for conductor commands to use the beta depot

#### Motivation:
For PROS 4 beta testing

##### References (optional):
Contains changes made in PR #253 and PR #251 and closes their respective issues, #243, #241

#### Test Plan:
- [x] Making project with ``--pros-4`` should use beta depot
- [x] Upgrading with ``pros c upgrade --pros-4`` on a PROS 3 project should have a warning
- [x] Downgrading with ``pros c apply-template kernel@3.x.x`` on a PROS 4 project should have a warning
- [x] ``pros c q --pros-4`` should list beta templates too

### CODE REVIEW

**Comment by ayushuk** [comment]
Will be revisited when the pros 4 is up on the depot. Was tested with dummy pros 3 template with adjusted metadata.

File: pros/conductor/conductor.py [hunk 1]
```diff
@@ -16,27 +17,39 @@
 
 MAINLINE_NAME = 'pros-mainline'
 MAINLINE_URL = 'https://purduesigbots.github.io/pros-mainline/pros-mainline.json'
+BETA_NAME = 'pros-4-beta'
+BETA_URL = 'https://purduesigbots.github.io/pros-mainline/beta/pros-4-beta.json'
```
Comment (WillXuCodes) [inline pros/conductor/conductor.py]: can we name this just `kernel-beta-mainline.json`? just a more futureproof name. idea is we use this for all beta testing on the kernel.

**Review by WillXuCodes — DISMISSED** [review]
Overall functionality is good, just needs one minor naming change.

**Comment by omegaStag** [comment]
progress: making pros 3 and 4 projects works. even when beta templates are cached, making a pros 3 project will make a pros 3 project. the up/downgrade warning shows up when you try and apply kernel 3 to kernel 4 project and vis versa. query-templates shows only mainline templates without the --beta flag, shows both mainline and beta with --beta. pros c upgrade will try to put your proj kernel to 4, which it will, and then errors out when it tries to upgrade any incompatible tempaltes also on the project. having a mismatched kernel 3 or 4 with a 4 or 3 compatible template will cause you to be unable to uninstall the incompatible ones until you go back to the compatible kernel version.

**Comment by omegaStag** [comment]
update: added a check in project class so that if an incompatible template is already installed it will uninstall it instead of not uninstalling. this allows for pros 3 templates to be uninstalled from pros 4 projects and vis versa.

File: pros/cli/conductor.py [hunk 2]
```diff
@@ -163,7 +170,7 @@ def upgrade(ctx: click.Context, project: c.Project, query: c.BaseTemplate, **kwa
 @click.option('--remove-user', is_flag=True, default=False, help='Also remove user files')
 @click.option('--remove-empty-dirs/--no-remove-empty-dirs', 'remove_empty_directories', is_flag=True, default=True,
               help='Remove empty directories when removing files')
-@click.option('--no-make-clean', is_flag=True, default=True, help='Do not run make clean after removing')
+@click.option('--no-make-clean', is_flag=True, default=False, help='Do not run make clean after removing')
```
Comment (kunwarsahni01) [inline pros/cli/conductor.py]:
```suggestion
```
Should probably not be in this PR

File: pros/cli/terminal.py
Comment (kunwarsahni01) [inline pros/cli/terminal.py:1]: Discard Changes

**Review by kunwarsahni01 — APPROVED** [review]
FIWB

**Pull request merged**